### ISSUE TITLE
Elements must have sufficient color contrast

### ISSUE THREAD

**Issue by seanforyou23** [patternfly/patternfly-next#620]
**_Please see suggested updates for this issue in a comment below_**

Some of the patternfly-next component/utility demo pages exhibit this issue. To run the a11y reporter locally and see these issues in action either use npm scripts by following the steps below or install the [axe browser extension](https://chrome.google.com/webstore/detail/axe/lhdoppojpmngadmnindnejefpokejbdd) and test individual pages manually. There's Firefox flavors out there of the axe extension if that's more your style.

- install latest [chromedriver](http://chromedriver.chromium.org/downloads) and ensure its available on your system $PATH
- alternatively, macOS users can simply brew cask install chromedriver
- run npm run dev
- run npm run a11y

Details related to this accessibility violation can be found here: https://dequeuniversity.com/rules/axe/3.0/color-contrast

Below are the URLs that illustrate the problem as of today:

- ~~[/components/NavSystem/examples-full/](https://pf-next.com/components/nav/examples-full/)~~ _This issue is now captured in #978_
- ~~[/demos/PageLayout/examples-full/](https://pf-next.com/demos/pagelayout/examples-full/)~~ This is now a component: 
  - [/components/Page/examples/](https://pf-next.com/components/Page/examples/)
- ~~[/layouts/Page/examples-full/](https://pf-next.com/layouts/page/examples-full/)~~ _This issue is now captured in #725_

Our accessibility audit considers this issue: `serious`

Related to https://github.com/patternfly/patternfly-next/issues/541

To learn more about how to build and test locally, check out: https://github.com/patternfly/patternfly-next#testing-for-accessibility

@mceledonia is the visual design contact that would be able to help with answers regarding colors and color contrast. 

**As part of this issue, please complete the following items.** These will help reviewers know what issues were addressed in the PR, and also will help us keep our total a11y violation count down.

1. Run the a11y audit, copy the issues from the audit that were fixed in the PR, and paste them into the PR description. 
    - Instructions for running the audit are here: https://github.com/patternfly/patternfly-next#testing-for-accessibility
    - Issues that are returned in the terminal look like this. Find the issues you are addressing and copy the following information provided for them or take a screencap to include in the PR.
      ```
      1 error found in: /upgrade-examples/AlertUpgradeExamples/examples-full/

      1) Buttons must have discernible text -- https://dequeuniversity.com/rules/axe/3.1/button-name -- impact: critical
      
      -------html violation instances-------

      <button class="btn btn-link dropdown-toggle" type="button" id="dropdownKebabRight" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
        <span class="fa fa-ellipsis-v"></span>
      </button> 
      ```
2. Reduce the threshold by number of issues being addressed 
    - This update is made [here](https://github.com/patternfly/patternfly-next/blob/master/test-a11y/config.js#L2)
    - The new value should be derived by: 
      - subtracting the number of issues you are addressing in the PR (e.g. the line that looks like this in the info you copied from the audit report in item 1 above):
        `5 errors found in: /components/Toolbar/examples-full/`
      - from current violations (e.g. this line in the audit report, near the end):
        `20 violations in total`

**Comment by dgutride** [comment]
This link shows the issue clearly:  https://webaim.org/resources/contrastchecker/?fcolor=007bba&bcolor=f5f5f5

**Comment by LHinson** [comment]
Is this still an open issue? Last I understood, this might have been addressed. @dabeng @seanforyou23

**Comment by seanforyou23** [comment]
@LHinson yes this is still an issue. Last I check there are color contrast issues in the following paths

/components/NavSystem/examples-full/
/demos/PageLayout/examples-full/
/layouts/Page/examples-full/

**Comment by jgiardino** [comment]
Created issue https://github.com/patternfly/patternfly-next/issues/725 to capture updates needed for /layouts/Page/examples-full/

**Comment by jgiardino** [comment]
@kybaker @mceledonia 

I was chatting with the CSS army about how we can make the Page component accessible when it isn't combined with the BackgroundImage component.

The solution we came up with was to apply a background color to the `<body>` of the page so that the default styles of the masthead toolbar section are accessible when a background image isn't present. 

Current example of Page component:
![image](https://user-images.githubusercontent.com/21063328/50295645-80678300-0446-11e9-8ec3-6af5fe6f18a2.png)

Page component with background color on body:
![screen shot 2018-12-20 at 11 25 24 am](https://user-images.githubusercontent.com/21063328/50297102-fae5d200-0449-11e9-8e2c-c1aef3b423d4.png)

Something to keep in mind with this is that the color we define on the `<body>` would bleed through other sections on the page as shown above.

If this isn't ideal, an alternate solution is to add a modifier class that only applies a background color to the toolbar section of the masthead.

**Comment by jgiardino** [comment]
UPDATE based on recent discussion with @kybaker and the rest of the css army...

The current page design is implemented using separate components, Page and BackgroundImage, where there are sections on the Page component that are transparent or semi-transparent so that the background image shows through. 

Our goal is to support the following use cases for the Page component:

1. the background image fails to load
2. the consumer does not want to use a background image

For both cases, the color in the toolbar section of the masthead would fail color contrast check because the text color is dependent on the BackgroundImage component loading. Currently, the preferred design for when no background image is present isn't possible for use case 1, given the way the Page and BackgroundImage components are set up to work together. For this case, we can provide a fallback solution that passes the a11y check. Whereas for use case 2, we can provide an alternative design.

Updates 
- For use case 1, apply a background color to the `<body>` element (should be the variable equivalent of #4d5258). If the background image is present, this color isn't visible, but if the image fails to load, this color will bleed through certain sections of the Page component.
- For use case 2, include a modifier class that applies a background color to the masthead elements (should be the variable equivalent of #030303). This solution requires that the consumer know that this modifier is available and to apply it. If they don't include the BackgroundImage component and don't use the modifier, the page will look as it would in use case 1.

**Examples**
Based on the colors mentioned above, these are screencaps of our currently busted Page component showing how the background colors would show through certain sections on the page (ignore the weird layout in the masthead).

Use case 1: background image fails to load
![image](https://user-images.githubusercontent.com/21063328/50611915-2df67f80-0ea6-11e9-908b-46018154d4fe.png)


Use case 2: consumer doesn't want a background image and uses the modifier class
![image](https://user-images.githubusercontent.com/21063328/50612032-875eae80-0ea6-11e9-9a44-b41cd4e5ab3d.png)

**Comment by mcoker** [comment]
Looks like all of these issues are resolved, except for one violation in the Navigation component. I've captured the cause in https://github.com/patternfly/patternfly-next/issues/1693.

Can we close this one in favor of #1693?

**Comment by jgiardino** [comment]
Yes, thank you!
